packages/node/src/cli.js

```diff
@@ -140,18 +141,26 @@ const main = async (args = hideBin(process.argv)) => {
    *   entrypoint: string
    *   bin?: boolean
    *   root: string
+   *   'trust-root'?: boolean
    * }} argv
    * @returns {void}
    */
   const processEntrypointMiddleware = (argv) => {
     const { entrypoint } = argv
     argv.entrypoint = argv.bin
-      ? resolveBinScript(argv.entrypoint, { from: argv.root })
+      ? resolveExecutable(argv.entrypoint, { from: argv.root })
       : resolveEntrypoint(argv.entrypoint, argv.root)
     if (entrypoint !== argv.entrypoint) {
-      // note: this will print if the original entrypoint is a relative path; we
-      // may or may not want to continue displaying it for that specific case.
-      log.warning(`Resolved ${entrypoint} to ${argv.entrypoint}`)
+      log.warning(`Resolved ${hrPath(entrypoint)} → ${hrPath(argv.entrypoint)}`)
+    }
+
+    // TODO: determine if checking `root` for `node_modules` is necessary
+    argv['trust-root'] = !argv.entrypoint.includes('node_modules')
+    argv['trust-root'] = false
+    if (!argv['trust-root']) {
+      log.info(
+        `Entrypoint is in a ${hrPath('node_modules/')} directory and is considered untrusted`
+      )
```

Comment:
- [ ] pull out into function and just call it instead of adding this crap to the middleware